### ISSUE TITLE
Write PP3 before processing image, not after

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#2494]
Originally reported on Google Code with ID 2511

```
I have noticed that the pp3 file is saved at the end of the raw conversion. 

I think it should be saved just after choosing save image and before processing it
so that it will reflect the output even if in the middle of the process an user modifies
some parameters, not at the end.

Discovered the problem tweaking white balance whilst rawtherapee was saving a jpg,
reloaded the pp3 for another image and it was "wrong".

Thank you in advance.

Branch: default
Version: 4.1.65
Changeset: 73afcbeb4260
Compiler: cc 4.8.2
Processor: undefined
System: Linux
Bit depth: 64 bits
Gtkmm: V2.24.4
Build type: Release
Build flags:  -Wno-unused-result -Wno-aggressive-loop-optimizations -march=native -fopenmp
-Werror=unknown-pragmas -O3 -DNDEBUG
Link flags:   -march=native
OpenMP support: ON
MMAP support: ON


```

Reported by `fotogiuliopv` on 2014-09-16 20:06:52

**Comment by Beep6581** [comment]
This is still an issue. Enable "Save processing parameters with image" and start saving, then carry on making modifications in RT, e.g. change white balance. The PP3 is saved after the image is saved, which means the PP3 does not correspond to the saved image.

**Comment by Beep6581** [comment]
Still valid. Edit image, Ctrl+s, save image with PP3, then set exposure to -5 before saving finishes. The saved PP3 has
```
[Exposure]
Compensation=-5
```

**Comment by heckflosse** [comment]
I will take the issue

**Comment by heckflosse** [comment]
@Beep6581 I didn't test, because my machine is too fast to change values in editor while saving an image. The fix should work, please test. If it doesn't work, we can undo the fix

Ingo

**Comment by Beep6581** [comment]
@heckflosse unfortunately it's not fixed yet.

If you need to test, just enable all of the heaviest tools - retinex, noise reduction, etc. Then Ctrl+s and tick to save pp3 with image, and change some parameter as soon as you hit ok.

**Comment by Beep6581** [comment]
Fix confirmed.
Fixed in 14378f39.
